magefile.publish.go

```diff
@@ -4,171 +4,136 @@ package main
 	{OS: "ubuntu", Dist: "eoan"},
 	{OS: "ubuntu", Dist: "focal"},
+	{OS: "ubuntu", Dist: "hirsute"},
 
-	{OS: "debian", Dist: "jessie"},
 	{OS: "debian", Dist: "stretch"},
 	{OS: "debian", Dist: "buster"},
 	{OS: "debian", Dist: "bullseye"},
 }
 
-func getArch(distro Distro) (string, error) {
-	if distro.OS == "el" || distro.OS == "fedora" {
-		return "x86_64", nil
-	}
-
-	if distro.OS == "ubuntu" || distro.OS == "debian" {
-		return "amd64", nil
-	}
+// Walk through directory and collect file paths satisfying patterns.
+func walkMatch(root string, patterns []string) ([]string, error) {
+	var matches []string
 
-	return "", fmt.Errorf("Unknown OS: %s", distro.OS)
-}
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 
-func getExt(distro Distro) (string, error) {
-	if distro.OS == "el" || distro.OS == "fedora" {
-		return rpmExt, nil
-	}
+		if info.IsDir() {
+			return nil
+		}
 
-	if distro.OS == "ubuntu" || distro.OS == "debian" {
-		return debExt, nil
-	}
+		for _, pattern := range patterns {
+			if matched, err := filepath.Match(pattern, filepath.Base(path)); err != nil {
+				return err
+			} else if matched {
+				matches = append(matches, path)
+				return nil
+			}
+		}
 
-	return "", fmt.Errorf("Unknown OS: %s", distro.OS)
-}
+		return nil
+	})
 
-func getPackagePath(distro Distro) (string, error) {
-	ext, err := getExt(distro)
 	if err != nil {
-		return "", fmt.Errorf("Failed to get ext: %s", err)
+		return nil, err
 	}
 
-	arch, err := getArch(distro)
-	if err != nil {
-		return "", fmt.Errorf("Failed to get arch: %s", err)
-	}
+	return matches, nil
+}
 
-	var packageNamePattern string
-	if ext == "deb" {
-		packageNamePattern = fmt.Sprintf("%s_*_%s.deb", packageName, arch)
-	} else if ext == "rpm" {
-		packageNamePattern = fmt.Sprintf("%s-*.%s.rpm", packageName, arch)
-	} else {
-		return "", fmt.Errorf("Unknown extension: %s", ext)
-	}
+// Patterns to select goreleaser build artifacts.
+// Based on update_repo.sh with some modifications.
+// See https://github.com/tarantool/tarantool/blob/c9e3926e5ae82785c5eb494521c44ba522109e95/tools/update_repo.sh#L517
+// and https://github.com/tarantool/tarantool/blob/c9e3926e5ae82785c5eb494521c44ba522109e95/tools/update_repo.sh#L986
+func getPatterns(distro Distro) ([]string, error) {
 
-	matches, err := filepath.Glob(filepath.Join(distPath, packageNamePattern))
-	if err != nil {
-		return "", fmt.Errorf("Failed to find matched files: %s", err)
+	if distro.OS == "el" || distro.OS == "fedora" {
+		return []string{"*.x86_64.rpm", "*.noarch.rpm"}, nil
 	}
 
-	if len(matches) == 0 {
-		return "", fmt.Errorf("No matched packages found for %s", packageNamePattern)
-	} else if len(matches) > 1 {
-		return "", fmt.Errorf("Found multiple matched packages for %s: %v", packageNamePattern, matches)
+	if distro.OS == "ubuntu" || distro.OS == "debian" {
+		return []string{"*.deb", "*.dsc", "*.Linux.*.tar.*z"}, nil
```

Comment:
Hm. This tar.gz is not sources. Just executable in an archive? Where we should place it for a user? Where it is placed with current code?

Reply:
I think update_repo.sh published it as is

Reply:
Does it mean that it was deployed to some URL, which nobody knows? I don't see much sense in that.

Just attach to a release on github, maybe?

Reply:
I think it will be released on GitHub with goreleaser without changes

Reply:
So I would remove it from the pattern for RWS.

Reply:
Removed